src/nice-path-decomposition.cpp

```diff
@@ -0,0 +1,100 @@
+#include "nice-path-decomposition.h"
+#include "../tdlib/src/iter.hpp"
+
+std::vector<NicePathDecomposition::Bag> NicePathDecomposition::get_nice_bags() {
+    return _nice_bags;
+}
+
+void NicePathDecomposition::set_g(Graph &g) {
+    _g = g;
+}
+
+NicePathDecomposition::NicePathDecomposition(std::vector<std::vector<vertex_t>> &bags, Graph &g) : PathDecomposition(
+        bags, g) {
+    fill_nice_bags();
+}
+
+
+NicePathDecomposition::NicePathDecomposition(PathDecomposition pathDecomposition) : PathDecomposition(
+        pathDecomposition) {
+    fill_nice_bags();
+}
+
+
+bool NicePathDecomposition::Bag::operator<(Bag b) {
```

Comment:
I don't quite get what is it for, is it just to have some comparator or does it have a more sophisticated purpose? If it is just to be able to use sets, I suggest simply joining all fields into a tuple and comparing the tuples.